src/main/java/org/jvnet/hudson/test/RealJenkinsRule.java

```diff
@@ -457,6 +464,116 @@ public RealJenkinsRule prepareHomeLazily(boolean prepareHomeLazily) {
         return this;
     }
 
+    /**
+     * Use {@link #withFIPSEnabled(Path)} with default value of {@code target/dependency}
+     * @return
+     */
+    public RealJenkinsRule withFIPSEnabled() {
+        return withFIPSEnabled(Paths.get("target/dependency"));
+    }
+
+    /**
+     * Enable FIPS mode assuming all FIPS BouncyCastle jars (bc-fips.jar, bctls-fips.jar and bcpkix-fips.jar) are in the provided directory.
+     * This will set some System properties such :
+     * <ul>
+     *   <li>-Xbootclasspath with FIPS BouncyCastle jars</li>
+     *   <li>java.security.properties to a new temp file configuring the security provider to FIPS BouncyCastle</li>
+     *   <li>-Dsecurity.overridePropertiesFile=true</li>
+     *   <li>-Djavax.net.ssl.trustStoreType=PKCS12</li>
+     *   <li>-Djenkins.security.FIPS140.COMPLIANCE=true</li>
+     *   <li>-Dcom.redhat.fips=false</li>
+     * </ul>
+     * @param fipsLibrariesPath path to directory with FIPS BouncyCastle jars
+     * @return
+     */
+    public RealJenkinsRule withFIPSEnabled(@NonNull Path fipsLibrariesPath) {
+        this.fipsLibrariesPath = Objects.requireNonNull(fipsLibrariesPath);
+        // check expected BouncyCastle fips jars are here
+        if  (!Files.exists(this.fipsLibrariesPath) ||
```

Comment:
another option would have been to use maven-resolver API and download files for the user.  But this would mean we force the jars version here.

Reply:
the java security and version of the jars may well be intertwined, so I am not sure that forcing the versions here would be a bad thing.
Additionally with the current setup all users need to duplicate the BC FIPS libraries in their pom config?

Reply:
Yes but at least users have the freedom to eventually test some upgrades to new version (note it's in the plugin configuration so hopefully dependabot or other tools will not propose any upgrade)
The problem with using something to download dependencies (because dependencies may not be in the local repository) is that all the setup must be available here (such as the path to the used settings.xml, which can be different from `~/.m2/settings.xml`), cli request properties, the content of  `.mvn/maven.config`)
That makes something complicated to achieve and maintain (as Maven3 and Maven4 might have differences) for minimal gain. 
Here, we provide a pom snippet to copy/paste. We can upgrade the pom snippet if we need to update the recommended versions.